tests/test_inner.py

```diff
@@ -848,3 +867,59 @@ def test_workflow_plugin_results():
     assert workflow.postbuild_results == {'post_build_value': 'post_build_value_result'}
     assert workflow.prepub_results == {'pre_publish_value': 'pre_publish_value_result'}
     assert workflow.exit_results == {'exit_value': 'exit_value_result'}
+
+
+def test_cancel_build(request):
+    """
+    Verifies that exit plugins are executed when the build is cancelled
+    """
+    this_file = inspect.getfile(PreRaises)
+    mock_docker()
+    fake_builder = MockInsideBuilder()
+    flexmock(InsideBuilder).new_instances(fake_builder)
+
+    watch_pre = Watcher()
+    watch_prepub = Watcher()
+    watch_post = WatcherWithPause()
```

Comment:
Would be nice if we could parametrize this to have the pause in each of these three phases and check that all phases up to the one we sent the signal to completed, that the one we sent the signal to did not complete, and that the exit phase did complete.

I guess this doesn't test what happens when the actual build step receives the signal but that will be easier once it is its own plugin phase.